rest-service/manager_rest/rest/resources_v3_1/status.py

```diff
@@ -0,0 +1,175 @@
+            syncthing_status = self._check_syncthing(services)
+
+        status = self._get_manager_status(systemd_statuses, rabbitmq_status,
+                                          syncthing_status)
+
+        # If the response should be only the summary - mainly for LB
+        if summary_response:
+            return {'status': status, 'services': {}}
+
+        return {'status': status, 'services': services}
+
+    def _check_systemd_services(self, services):
+        systemd_services = get_services(
+            get_systemd_manager_services(BASE_SERVICES, OPTIONAL_SERVICES)
+        )
+        statuses = {ACTIVE_STATE}
+        for service in systemd_services:
+            if should_be_in_services_output(service, OPTIONAL_SERVICES):
+                status = ACTIVE_STATE if service['instances'][0]['state'] == \
+                                         'running' else INACTIVE_STATE
+                services[service['display_name']] = {
+                    'status': status,
+                    'is_external': False,
+                    'extra_info': {
+                        'systemd': service
+                    }
+                }
+                statuses.add(status)
+        return statuses
+
+    def _check_rabbitmq(self, services):
+        broker_status = ACTIVE_STATE if broker_is_healthy() else INACTIVE_STATE
+        self._add_or_update_service(services, 'RabbitMQ', broker_status)
+        return broker_status
+
+    def _check_syncthing(self, services):
+        status = INACTIVE_STATE
+        display_name = 'File Sync Service'
+
+        try:
+            syncthing_config = syncthing_utils.config()
+            device_stats = syncthing_utils.device_stats()
+        except Exception as err:
+            current_app.logger.error(
+                'Syncthing check failed with {err_type}: {err_msg}'.format(
+                    err_type=type(err),
+                    err_msg=str(err),
+                )
+            )
+            self._add_or_update_service(services, display_name, status)
+            return status
+
+        interval = syncthing_config['options']['reconnectionIntervalS'] + 1
+        min_last_seen = datetime.utcnow() - timedelta(seconds=interval)
+
+        for device_id, stats in device_stats.items():
+            last_seen = parse_datetime_string(stats['lastSeen'])
+
+            # Syncthing is valid when at least one device was seen recently
+            if last_seen > min_last_seen:
+                status = ACTIVE_STATE
+                break
+
+        self._add_or_update_service(services, display_name, status)
+        return status
+
+    def _get_manager_status(self, systemd_statuses, rabbitmq_status,
+                            syncthing_status):
+        statuses = systemd_statuses.union({rabbitmq_status, syncthing_status})
+
+        # statuses set can maximum have 2 statuses Active and Inactive
+        return HEALTHY_STATE if len(statuses) == 1 and \
```

Comment:
Maybe better as: return HEALTHY_STATE if all(status == HEALTHY_STATE for status in statuses) else FAIL_STATE
?

Reply:
But then I don't use the advantage of the set

Reply:
I think it's more readable with the all(x) approach though? Is there somewhere else that we get significant advantage from making this a set?